### ISSUE TITLE
[BUG] error bars in plot_pull

### ISSUE THREAD

**Issue by aminnj** [scikit-hep/hist#146]
Should https://github.com/scikit-hep/hist/blob/221d28ffd09fd1dadb87be9306c56146f4095f8b/src/hist/plot.py#L173 have a sqrt? I.e.,
```python
yerr = self.variances() ** 0.5
```
Otherwise, the plot/pulls look to be off for this example.

```python
import numpy as np
from hist import Hist
from uncertainties import unumpy as unp

np.random.seed(42)
hh = Hist.new.Reg(50, -5, 5).Double().fill(np.random.normal(0,1,int(1e5)))

def func(x, constant, mean, sigma):
    exp = unp.exp if constant.dtype == np.dtype("O") else np.exp
    return constant * exp(-((x - mean) ** 2.0) / (2 * sigma ** 2))

hh.plot_pull(func)
```

![image](https://user-images.githubusercontent.com/5760027/111009669-274ad900-8349-11eb-8e61-2add7887ad3b.png)

Changing it wrt master gives me this.

![image](https://user-images.githubusercontent.com/5760027/111009793-709b2880-8349-11eb-9f67-9eecf34ccbf9.png)

Also, I'm wondering if there would be a good reason to not add a little bit more sugar to the fitting. E.g., I could contribute a wrapper around `curve_fit` which recognizes `a=1` as a default and feeds it into `curve_fit`'s `p0`
```python
def f(x, a=1):
    return a*x
```
as in https://github.com/aminnj/yahist/blob/dcbcd055e701775f820e97c95e66599b726530eb/yahist/utils.py#L249.
I also found the switching between `unp` (`uncertainties.numpy`) and `np` in the user callable to be a little unintuitive. Would you consider having imuinuit as a dependency instead of `uncertainties`? Then it's only a couple of extra lines to get the replacement for `curve_fit`'s `pcov` from `iminuit` and resample to get the error band
```python
            vopts = np.random.multivariate_normal(popt, pcov, nsamples)
            sampled_ydata_fine = np.vstack(
                [func(xdata_fine, *vopt).T for vopt in vopts]
            )
```

Let me know if I should move this to a separate issue/discussion :)

**Comment by henryiii** [comment]
Looks like a bug, thanks!

And for your second point, I'd rather like to move to iminuit and drop the SciPy requirement entirely (except for now through mplhep). Happy to have help if you want to. ;)

**Comment by aminnj** [comment]
Great. I'll find some time to make a PR then. :)

**Comment by matthewfeickert** [comment]
Was about to report this, so glad to see this is in PR #149.

**Comment by henryiii** [comment]
I'm going to fix the bug for 2.2.1, then get the PR in for 2.3. :) Currently waiting on WeightedMean in https://github.com/scikit-hep/boost-histogram/pull/537, then will be pushing new versions.

**Comment by matthewfeickert** [comment]
From my own use of `hist` `v2.2.1` I can say that this got resolved in PR #150, so I'm closing this Issue. Please reopen if this was missed elsewhere.

**Comment by henryiii** [comment]
This was a two part issue, but the other part was done in #149, so yes, perfectly closable now. :)
